bpf/init.sh

```diff
@@ -471,14 +466,19 @@ case "${MODE}" in
 		echo "#define EPHEMERAL_MIN $CILIUM_EPHEMERAL_MIN" >> $RUNDIR/globals/node_config.h
 
 		if [ "$NODE_PORT" = "true" ]; then
-			sed -i '/^#.*NATIVE_DEV_IFINDEX.*$/d' $RUNDIR/globals/node_config.h
-			NATIVE_DEV_IDX=$(cat /sys/class/net/${NATIVE_DEV}/ifindex)
-			echo "#define NATIVE_DEV_IFINDEX $NATIVE_DEV_IDX" >> $RUNDIR/globals/node_config.h
-			sed -i '/^#.*NATIVE_DEV_MAC.*$/d' $RUNDIR/globals/node_config.h
-			NATIVE_DEV_MAC=$(ip link show $NATIVE_DEV | grep ether | awk '{print $2}')
-			NATIVE_DEV_MAC=$(mac2array $NATIVE_DEV_MAC)
-			echo "#define NATIVE_DEV_MAC { .addr = ${NATIVE_DEV_MAC}}" >> $RUNDIR/globals/node_config.h
-
+			MAC_BY_IFINDEX_MACRO="#define NATIVE_DEV_MAC_BY_IFINDEX(IFINDEX) ({ \\
```

Comment:
I guess we won't support hot-plugging these.

(It's in theory supportable, someone would just need to convince `cilium-agent` to fully recompile all the BPF progs)

Reply:
Are you saying that even if a user restarts cilium-agent with different list of `--device`, some programs will run with the old list of devs?

Reply:
I was thinking something more like EKS environments with ENI devices being created / destroyed during a single Cilium agent runtime. Unless we reconfigure these macros, we wouldn't pick up the new devices.

Not sure whether it's possible for such ENIs to be configured and also provide access to the outside world so unless that's a clear use case, it's probably not worth worrying about for now.